Content: Add maskable icon post (#1975)
GoogleChrome/web.dev: src/site/content/en/blog/maskable-icon/index.md

#     alt: Provide an alt for your thumbnail.
#   facebook:
#     title: A title for Facebook card.
#     description: A description for Facebook card.
#     thumbnail: facebook_thumbnail.jpg
#     alt: Provide an alt for your thumbnail.
#   twitter:
#     title: A title for Twitter card.
#     description: A description for Twitter card.
#     thumbnail: twitter_thumbnail.jpg
#     alt: Provide an alt for your thumbnail.

tags:
  - post
  - capabilities
  - progressive-web-apps
---

## What are maskable icons? {: #what }

If you've installed a Progressive Web App on a recent Android phone, you might notice the icon shows up with a white background. {{image}}. Android Oreo introduced adaptive icons, which display app icons in a variety of shapes across different device models. Icons that don't follow this new format are given white backgrounds.

Maskable icons are a new icon format that let your Progressive Web App use adaptive icons. If you supply a maskable icon, your icon can fill up the entire shape and look great on all Android devices. Firefox and Chrome have recently added support for this new format, and you can adopt it in your apps.

## Are my current icons ready?

> Since maskable icons need to support a variety of shapes, you supply an opaque image with some padding that the browser can later crop into the correct shape and size. The important parts of your icon, such as your logo, should be within the middle 80% of the icon. The outer 10% edge may be cropped.

Reviewer: ```suggestion
Since maskable icons need to support a variety of shapes, you supply an opaque image with some padding that the browser can later crop into the desired shape and size. It's best not to rely on any particular shape, since it can vary by browser and per platform.

Luckily, there's a well-defined and [standardized](https://w3c.github.io/manifest/#icon-masks) "minimum safe zone" that all browsers respect. The important parts of your icon, such as your logo, should be within the middle 80% of the icon. The outer 10% edge may be cropped.
```

> Since maskable icons need to support a variety of shapes, you supply an opaque image with some padding that the browser can later crop into the correct shape and size. The important parts of your icon, such as your logo, should be within the middle 80% of the icon. The outer 10% edge may be cropped.

Reviewer 2: > should be within the middle 80% of the icon. The outer 10% edge may be cropped.

This is a little confusing, since you're kind of "mixing units" in a way. The "middle 80%" is kind referring to the diameter of the icon, while the "outer 10% edge" is referring to the radius. If you don't already understand what's going on, it's weird that these don't add up to 100%.

> You can check which parts of your icons land within the safe zone with Chrome DevTools. With your Progressive Web App open, open DevTools and navigate to the "Application" panel. In the "Icons" section, you can choose to "Show only the minimum safe area for maskable icons". {{image}}. Your maskable icons will be trimmed so that only the safe area is visible. If your logo is visible within this area, you're good to go.

Reviewer 2: I would suggest calling this reasoning out specifically in the blog post.

If you want to preview your maskable icon in other shapes it may appear in on Android, I've created a tool called [maskable.app](https://maskable.app/). Upload an icon and maskable.app will let you try out various shapes and sizes, and you can share the preview with others on your team. {{image}}.

## How do I adopt maskable icons?

If you want to create a maskable icon based off your existing transparent icon, you can use the [maskable.app editor](https://maskable.app/editor). Upload your icon, adjust the color and size, then export the image. {{image}}.

> Once you've created a maskable icon image and tested it out in DevTools, you'll need to update your Web App Manifest to point to the new assets. The Web App Manifest provides information about your web app in a JSON file, and includes an `icons` array. {{code}}.

Reviewer: Opportunity to link to https://developers.google.com/web/fundamentals/web-app-manifest. cc @PaulKinlan @gauntface, are there any plans to port this to web.dev?

> With the inclusion of maskable icons, a new property has been added for image resources listen in a Web App Manifest. The `purpose` field tells the browser how your icon should be used. By default, icons will have a purpose of `"any"` to indicate they can be used for any launcher icon. {{code}}.

Reviewer 2: > a new property has been added for image resources listen in a Web App Manifest

s/listen/listed/

> Maskable icons should use a different purpose: `"maskable"`. This indicates that an image is meant to be used with icon masks. You can also specify multiple purposes, if you want your maskable icon to be used without a mask on other devices. {{code}}.

Reviewer: Let's clarify the difference between `any` and `maskable` here. The previous sentence implies that `any` already includes `maskable`, which is true in the sense that the icon could first be resized by the browser and _then_ made maskable. The benefit of marking an icon as `maskable` is that it gives you more fine-grained control over the result: maskable icons skip the browser-dependent resizing step.

---
title: Better PWA icons with maskable icons
subhead: A new icon format to use adaptive icons on supporting platforms.
description: |
  A description of the article that will appear in search results.
authors:
  - tigeroakes
date: 2019-12-09
tags:
  - post
  - capabilities
  - progressive-web-apps
---

## What are maskable icons? {: #what }

<figure class="w-figure w-figure--inline-right">
  <video class="w-screenshot" autoplay loop muted playsinline aria-label="Different masks applied to a maskable icon">
    <source src="fugu-mask.webm" type="video/webm; codecs=vp8">
    <source src="fugu-mask.mp4" type="video/mp4; codecs=h264">
  </video>

> </figure>

Reviewer: I like this animation, but to make it extra clear what maskable icons are about we could show some home screen screenshots. Kind of like how @petele did it in his video here: https://www.youtube.com/watch?v=L0OB0_bO5I0&t=24s @petele Do you still have these image resources and could we use them here?

Author: I added my own variant of those resources as the hero. Do you think I should remove the animation or keep both the hero and video?

![Icons contained inside white circles compared to icons covering its entire circle](https://user-images.githubusercontent.com/1782266/70840400-d905fd80-1dc6-11ea-83be-fad1c8f769af.png)

Reviewer: That looks great!

The hero is not showing up for me on [the preview URL](https://deploy-preview-1975--web-dev-staging.netlify.com/maskable-icon/).

Reviewer 3: It's showing now.

Reviewer: **Update:** Never mind, it does now. Caching is hard? :D

Reviewer: IMHO we should repeat the relevant parts of the hero image in the body, in addition to your animation. Something like this:

```
Without explicit maskable icons, your PWA icons might look like this:

insert top part of <https://user-images.githubusercontent.com/81942/70888356-70926880-1fe0-11ea-8a12-75cf41e3051a.png>

With maskable icons, you have more control etc. etc.

insert bottom part of <https://user-images.githubusercontent.com/81942/70888356-70926880-1fe0-11ea-8a12-75cf41e3051a.png>

Note that the shape is platform-specific: it could be a circle, a squircle, etc. etc.

insert your animation
```

Author: Updated!

---
title: Better PWA icons with maskable icons

> subhead: A new icon format to use adaptive icons on Android.

Reviewer 3: ```suggestion
subhead: A new icon format to use adaptive icons on supporting platforms.
```

I wouldn't limit this to Android alone.

  <video class="w-screenshot" autoplay loop muted playsinline aria-label="Different masks applied to a maskable icon">
    <source src="fugu-mask.webm" type="video/webm; codecs=vp8">
    <source src="fugu-mask.mp4" type="video/mp4; codecs=h264">
  </video>
</figure>

If you've installed a Progressive Web App on a recent Android phone, you might notice the icon shows up with a white background. Android Oreo introduced adaptive icons, which display app icons in a variety of shapes across different device models. Icons that don't follow this new format are given white backgrounds.

Maskable icons are a new icon format that let your Progressive Web App use adaptive icons. If you supply a maskable icon, your icon can fill up the entire shape and look great on all Android devices. Firefox and Chrome have recently added support for this new format, and you can adopt it in your apps.

## Are my current icons ready?

Since maskable icons need to support a variety of shapes, you supply an opaque image with some padding that the browser can later crop into the desired shape and size. It's best not to rely on any particular shape, since it can vary by browser and per platform.

> Luckily, there's a well-defined and [standardized](https://w3c.github.io/manifest/#icon-masks) "minimum safe zone" that all browsers respect. The important parts of your icon, such as your logo, should be within a circular area in the center of the icon with a radius equal to 40% of the icon width. The outer 10% edge may be cropped.

Reviewer 3: ```suggestion
Luckily, there's a well-defined and [standardized](https://w3c.github.io/manifest/#icon-masks) "minimum safe zone" that all platforms respect. The important parts of your icon, such as your logo, should be within a circular area in the center of the icon with a radius equal to 40% of the icon width. The outer 10% edge may be cropped.
```

I think "platforms" is the better word here, since the icon lands on the home screen.

> You can check which parts of your icons land within the safe zone with Chrome DevTools. With your Progressive Web App open, open DevTools and navigate to the "Application" panel. In the "Icons" section, you can choose to "Show only the minimum safe area for maskable icons". Your icons will be trimmed so that only the safe area is visible. If your logo is visible within this area, you're good to go.

Reviewer 3: ```suggestion
You can check which parts of your icons land within the safe zone with Chrome DevTools. With your Progressive Web App open, launch DevTools and navigate to the "Application" panel. In the "Icons" section, you can choose to "Show only the minimum safe area for maskable icons". Your icons will be trimmed so that only the safe area is visible. If your logo is visible within this area, you're good to go.
```

Avoid repeated "open".

<figure class="w-figure">

>   <img src="devtools.png" alt="" style="max-width: 500px;">

Reviewer: As a side note, we're currently improving the way this looks for smaller icons, so we'll want to redo these screenshots before publication.

  <figcaption class="w-figcaption">
    The Applications panel
  </figcaption>
</figure>

If you want to preview your maskable icon in other shapes it may appear in on Android, I've created a tool called [Maskable.app](https://maskable.app/). Open an icon and Maskable.app will let you try out various shapes and sizes, and you can share the preview with others on your team.

> 

Reviewer 3: Nit: remove.

## How do I adopt maskable icons?

> If you want to create a maskable icon based off your existing transparent icon, you can use the [Maskable.app Editor](https://maskable.app/editor). Upload your icon, adjust the color and size, then export the image.

Reviewer 3: ```suggestion
If you want to create a maskable icon based off your existing icon, you can use the [Maskable.app Editor](https://maskable.app/editor). Upload your icon, adjust the color and size, then export the image.
```

Or does it indeed have to be a transparent version? I don't think so, but am happy to be corrected.

Meta question for the editor: Can it or should it be able to add a transparent layer?

Author: The editor can add a transparent layer. It forces an opaque background layer so you must export a non-transparent icon.

<figure class="w-figure">

>   <img src="maskable-app-editor.png" alt="" style="max-width: 500px;">

Reviewer 3: Nit: if the default out-of-the-box experience for the app is light mode, it should be depicted in light mode.

  <figcaption class="w-figcaption">
    Creating icons in Maskable.app Editor
  </figcaption>
</figure>

Once you've created a maskable icon image and tested it out in DevTools, you'll need to update your [Web App Manifest](https://developers.google.com/web/fundamentals/web-app-manifest) to point to the new assets. The Web App Manifest provides information about your web app in a JSON file, and includes an [`icons` array](https://developers.google.com/web/fundamentals/web-app-manifest#icons).

> With the inclusion of maskable icons, a new property has been added for image resources listed in a Web App Manifest. The `purpose` field tells the browser how your icon should be used. By default, icons will have a purpose of `"any"`. These icons resized on top of a white background by the browser on Android.

Reviewer 3: ```suggestion
With the inclusion of maskable icons, a new property value has been added for image resources listed in a Web App Manifest. The `purpose` field tells the browser how your icon should be used. By default, icons will have a purpose of `"any"`. These icons will be resized on top of a white background on Android.
```

The property `"purpose"` was already there, but a new valid value `"maskable"` was added.

I think the last sentence was missing "will be", and I am not sure if the resizing happens on the browser, or rather the OS level.

> Maskable icons should use a different purpose: `"maskable"`. This indicates that an image is meant to be used with icon masks, giving you more control over the result. This way, your icons will not have a white background. You can also specify multiple purposes, if you want your maskable icon to be used without a mask on other devices.

Reviewer 3: ```suggestion
Maskable icons should use a different purpose: `"maskable"`. This indicates that an image is meant to be used with icon masks, giving you more control over the result. This way, your icons will not have a white background. You can also specify multiple space-separated purposes (for example, `"any maskable"`), if you want your maskable icon to be used without a mask on other devices.
```

```json
{
  ...
  "icons": [
    ...
    {
      "src": "path/to/maskable_icon.png",
      "sizes": "196x196",
      "type": "image/png",

>       "purpose": "any maskable" // <-- New property

Reviewer 3: ```suggestion
      "purpose": "any maskable" // <-- New property value `"maskable"`
```

    }
  ]
  ...
}
```

> With this, you can go forth and create your own maskable icons, making sure you app looks great edge-to-edge.

Reviewer 3: ```suggestion
With this, you can go forth and create your own maskable icons, making sure you app looks great edge-to-edge (and for what it's worth, circle to circle, oval to oval 😄).
```

Feel free to ignore if you think that's too cheesy.

---
title: Better PWA icons with maskable icons
subhead: A new icon format to use adaptive icons on supporting platforms.
description: |

>   A description of the article that will appear in search results.

Reviewer 4: Needs a description.

authors:
  - tigeroakes
date: 2019-12-13
hero: hero.png
alt: Icons contained inside white circles compared to icons covering its entire circle
tags:
  - post
  - capabilities
  - progressive-web-apps
---

## What are maskable icons? {: #what }

If you've installed a Progressive Web App on a recent Android phone, you might notice the icon shows up with a white background. Android Oreo introduced adaptive icons, which display app icons in a variety of shapes across different device models. Icons that don't follow this new format are given white backgrounds.

> <figure class="w-figure">

Reviewer 4: You may also want to add the `w-figure--center` class to center these images. That seems to look a bit nicer.

>   <img src="homescreen-any.png" alt="" style="max-width: 400px;">

Reviewer 4: Need an alt description for this and all images. Using `alt=""` hides the image from screen readers, but they would still hear the figcaption, which could be confusing.

Although you're providing a caption, unfortunately we can't use that as a replacement for alt text.
https://www.scottohara.me/blog/2019/01/21/how-do-you-figure.html#a-figcaption-is-not-a-substitute-for-alt-text

I usually suggest to think of alt text as if you were describing the image to someone over the phone. What are the key visual bits you'd want to convey. The caption, on the other hand, provides additional context that helps explain how the image relates to the rest of the document.

  <figcaption class="w-figcaption">
    PWA icons covering the entire circle on Android
  </figcaption>
</figure>

## Are my current icons ready?

Since maskable icons need to support a variety of shapes, you supply an opaque image with some padding that the browser can later crop into the desired shape and size. It's best not to rely on any particular shape, since it can vary by browser and per platform.

<figure class="w-figure w-figure--inline-right">
  <video class="w-screenshot" autoplay loop muted playsinline aria-label="Different masks applied to a maskable icon">
    <source src="fugu-mask.webm" type="video/webm; codecs=vp8">
    <source src="fugu-mask.mp4" type="video/mp4; codecs=h264">
  </video>
  <figcaption class="w-figcaption">
    Different platform specific shapes
  </figcaption>
</figure>

Luckily, there's a well-defined and [standardized](https://w3c.github.io/manifest/#icon-masks) "minimum safe zone" that all platforms respect. The important parts of your icon, such as your logo, should be within a circular area in the center of the icon with a radius equal to 40% of the icon width. The outer 10% edge may be cropped.

> You can check which parts of your icons land within the safe zone with Chrome DevTools. With your Progressive Web App open, launch DevTools and navigate to the "Application" panel. In the "Icons" section, you can choose to "Show only the minimum safe area for maskable icons". Your icons will be trimmed so that only the safe area is visible. If your logo is visible within this area, you're good to go.

Reviewer 3: ```suggestion
You can check which parts of your icons land within the safe zone with Chrome DevTools. With your Progressive Web App open, launch DevTools and navigate to the "Application" panel. In the "Icons" section, you can choose to "Show only the minimum safe area for maskable icons". Your icons will be trimmed so that only the safe area is visible. If your logo is visible within this safe area, you're good to go.
```

Super nit: add a filler word to avoid a lone "go" in the new line.

<figure class="w-figure">

>   <img src="devtools.png" alt="" style="max-width: 500px;">

Reviewer 4: ```suggestion
  <img src="devtools.png" class="w-screenshot" alt="" style="max-width: 500px;">
```

Reviewer 4: You may also want to increase the size of this image or remove the max-width entirely to let it be full width. I found the text a little hard to read.

It may also look nicer centered with the `w-figure--center` class.

  <figcaption class="w-figcaption">
    The Applications panel
  </figcaption>
</figure>

> If you want to preview your maskable icon in other shapes it may appear in on Android, I've created a tool called [Maskable.app](https://maskable.app/). Open an icon and Maskable.app will let you try out various shapes and sizes, and you can share the preview with others on your team.

Reviewer 3: ```suggestion
If you want to preview your maskable icon in other shapes it may appear in on Android, I've created a tool called [Maskable.app](https://maskable.app/). Open an icon, then Maskable.app will let you try out various shapes and sizes, and you can share the preview with others on your team.
```

Just to separate the two phrases and avoid misreading as a list.

## How do I adopt maskable icons?

If you want to create a maskable icon based off your existing icon, you can use the [Maskable.app Editor](https://maskable.app/editor). Upload your icon, adjust the color and size, then export the image.

> <figure class="w-figure">

Reviewer 4: Suggest adding `w-figure--center`.

>   <img src="maskable-app-editor.png" alt="" style="max-width: 500px;">

Reviewer 4: ```suggestion
  <img src="maskable-app-editor.png" class="w-screenshot" alt="" style="max-width: 500px;">
```

---

> title: Better PWA icons with maskable icons

Reviewer 5: ```suggestion
title: Adaptive icon support in PWAs with maskable icons
```
You'll capture more SEO traffic by including `adaptive icons` in your title. Title wording seems to have a big impact on documentation SEO. This phrasing is also more precise.

I don't know this domain so you can ignore this if it's technically inaccurate.

subhead: A new icon format to use adaptive icons on supporting platforms.
description: |
  Maskable icons are a new icon format that give you more control and let your Progressive Web App
  use adaptive icons. By supplying a maskable icon, your icon can look great on all Android
  devices.
authors:
  - tigeroakes
date: 2019-12-19

> hero: hero.png

Reviewer 5: Please run `hero.png` through tinypng.com. I got it down to 200KB. This keeps the size of the repository itself down.

<figure class="w-figure w-figure--center">
  <img src="homescreen-maskable.png" alt="PWA icons covering the entire circle on Android" style="max-width: 400px;">
  <figcaption class="w-figcaption">
    Maskable icons cover the entire circle instead
  </figcaption>
</figure>

## Are my current icons ready?

Since maskable icons need to support a variety of shapes, you supply an opaque image with some padding that the browser can later crop into the desired shape and size. It's best not to rely on any particular shape, since it can vary by browser and per platform.

<figure class="w-figure w-figure--inline-right">
  <video class="w-screenshot" autoplay loop muted playsinline aria-label="Different masks applied to a maskable icon">
    <source src="fugu-mask.webm" type="video/webm; codecs=vp8">
    <source src="fugu-mask.mp4" type="video/mp4; codecs=h264">
  </video>
  <figcaption class="w-figcaption">
    Different platform specific shapes
  </figcaption>

> </figure>

Reviewer 5: Great use of `inline-right`! The post looks super snazzy on desktop.

Luckily, there's a well-defined and [standardized](https://w3c.github.io/manifest/#icon-masks) "minimum safe zone" that all platforms respect. The important parts of your icon, such as your logo, should be within a circular area in the center of the icon with a radius equal to 40% of the icon width. The outer 10% edge may be cropped.

> You can check which parts of your icons land within the safe zone with Chrome DevTools. With your Progressive Web App open, launch DevTools and navigate to the "Application" panel. In the "Icons" section, you can choose to "Show only the minimum safe area for maskable icons". Your icons will be trimmed so that only the safe area is visible. If your logo is visible within this safe area, you're good to go.

Reviewer 5: ```suggestion
You can check which parts of your icons land within the safe zone with Chrome DevTools. With your Progressive Web App open, launch DevTools and navigate to the **Application** panel. In the **Icons** section, you can choose to **Show only the minimum safe area for maskable icons**. Your icons will be trimmed so that only the safe area is visible. If your logo is visible within this safe area, you're good to go.
```

UI elements should be bold: https://developers.google.com/style/ui-elements#formatting

    Creating icons in Maskable.app Editor
  </figcaption>
</figure>

Once you've created a maskable icon image and tested it out in DevTools, you'll need to update your [Web App Manifest](https://developers.google.com/web/fundamentals/web-app-manifest) to point to the new assets. The Web App Manifest provides information about your web app in a JSON file, and includes an [`icons` array](https://developers.google.com/web/fundamentals/web-app-manifest#icons).

With the inclusion of maskable icons, a new property value has been added for image resources listed in a Web App Manifest. The `purpose` field tells the browser how your icon should be used. By default, icons will have a purpose of `"any"`. These icons will be resized on top of a white background on Android.

Maskable icons should use a different purpose: `"maskable"`. This indicates that an image is meant to be used with icon masks, giving you more control over the result. This way, your icons will not have a white background. You can also specify multiple space-separated purposes (for example, `"any maskable"`), if you want your maskable icon to be used without a mask on other devices.

```json
{

>   ...

Reviewer 5: ```suggestion
  …
```
We use ellipsis character rather than 3 periods because assistive technology announces the ellipsis character more accurately

  "icons": [

>     ...

Reviewer 5: ```suggestion
    …
```

    {
      "src": "path/to/maskable_icon.png",
      "sizes": "196x196",
      "type": "image/png",
      "purpose": "any maskable" // <-- New property value `"maskable"`
    }
  ]

>   ...

Reviewer 5: ```suggestion
  …
```

}
```

> With this, you can go forth and create your own maskable icons, making sure you app looks great edge-to-edge (and for what it's worth, circle to circle, oval to oval 😄).

Reviewer 5: Love it, great ending
